main.go

```diff
@@ -115,6 +116,39 @@ type ObjRef struct {
 	Offloaded uint
 }
 
+// updateAutoretrieveIndex ticks every tickInterval and checks for new information to add to autoretrieve
+// If so, it updates the filecoin index with the new CIDs, saying they are present on autoretrieve
+// With that, clients using bitswap can query autoretrieve servers using bitswap and get data from estuary
+func (s *Server) updateAutoretrieveIndex(tickInterval time.Duration, quit chan struct{}) error {
+	var autoretrieves []Autoretrieve
+	var lastTickTime time.Time
+	ticker := time.NewTicker(tickInterval)
+
+	for {
+		lastTickTime = time.Now().UTC().Add(-tickInterval)
+
+		// Find all autoretrieve servers that are online (that sent heartbeat)
+		err := s.DB.Find(&autoretrieves, "last_connection > ?", lastTickTime).Error
+		if err != nil {
+			return err
```

Comment:
I am not sure we want to quit if no `autoretrieves` were found. Only quit for a non `ErrRecordNotFound` error

Reply:
I don't think `Find` throws that error, from [here](https://gorm.io/docs/query.html):
> If you want to avoid the ErrRecordNotFound error, you could use Find like db.Limit(1).Find(&user), the Find method accepts both struct and slice data

Reply:
Yeah, Find will return an empty list, `First` will error if none are found

Reply:
I think that doc your referenced is for v2 of gorm(`Find` no longer returns this error in v2, but does in v1), and we are using v1, no? or better still can you verify that scenario?